client/app/states/vms/details/details.state.js

```diff
@@ -0,0 +1,166 @@
+      'tags',
+      'template',
+      'thin_provisioned',
+      'tools_status',
+      'uid_ems',
+      'users',
+      'vendor',
+      'vmsafe_enable',
+    ];
+    var options = {
+      attributes: requestAttributes,
+    };
+
+    return CollectionsApi.get('vms', $stateParams.vmId, options);
+  }
+  /** @ngInject */
+  function StateController($state, vmDetails, CollectionsApi, EventNotifications, Consoles) {
+    var vm = this;
+  
+    vm.vmDetails = vmDetails;
+    setInitialVars();
+
+    function setInitialVars() {
+      if (vm.vmDetails.cloud === true) {
+        resolveInstance(vm.vmDetails.id);
+      }
+
+      var neverText = __('Never');
+      var noneText = __('None');
+      var availableText = __('Available');
+      var notAvailable = __("Not Available");
+      vm.vmDetails.lastSyncOn = (angular.isUndefined(vm.vmDetails.last_sync_on) ? neverText : vm.vmDetails.last_sync_on);
+      vm.vmDetails.retiresOn = (angular.isUndefined(vm.vmDetails.retires_on) ? neverText : vm.vmDetails.retires_on);
+      vm.vmDetails.snapshotCount = defaultText(vm.vmDetails.snapshots);
+      vm.vmDetails.resourceAvailability = (vm.vmDetails.template === false ? availableText : noneText);
+      vm.vmDetails.driftHistory = defaultText(vm.vmDetails.drift_states);
+      vm.vmDetails.scanHistoryCount = defaultText(vm.vmDetails.scan_histories);
+      vm.vmDetails.lastComplianceStatus = (angular.isUndefined(vm.vmDetails.last_compliance_status) ? __('Never Verified') : vm.vmDetails.last_compliance_status);
+      vm.vmDetails.complianceHistory = (vm.vmDetails.compliances.length > 0 ? availableText : notAvailable);
+      var options = {
+        attributes: ['miq_request'],
+      };
+      CollectionsApi.get('services', vmDetails.service.id, options).then(serviceSuccess, {});
+    }
+    function defaultText(inputCount, defaultText) {
+      var inputArrSize = inputCount.length;
+      if (defaultText === null) {
+        defaultText = 'None';
+      }
+      if (inputArrSize === 0) {
+        return __(defaultText);
+      } else {
+        return inputArrSize;
+      }
+    }
+    function serviceSuccess(data) {
+      vm.vmDetails.provisionDate = data.miq_request.fulfilled_on;
+    }
+    function resolveInstance(instanceId) {
+      var requestAttributes = [
+        'availability_zone',
+        'cloud_networks',
+        'cloud_subnets',
+        'cloud_tenant',
+        'cloud_volumes',
+        'flavor',
+        'floating_ip_addresses',
+        'key_pairs',
+        'load_balancers',
+        'mac_addresses',
+        'network_ports',
+        'network_routers',
+        'miq_provision_template',
+        'orchestration_stack',
+        'security_groups',
+      ];
+      var options = {
+        attributes: requestAttributes,
+      };
+
+      CollectionsApi.get('instances', instanceId, options).then(processInstanceVariables, {});
+    }
+    function processInstanceVariables(data) {
+      var noneText = __('None');
```

Comment:
this is already declared in the controller, maybe increase the scope (as it is used in multiple functions)